### PR TITLE
🎉Source Okta: unit tests were added

### LINKED ISSUE #14825: Source Okta. Cover functionality with unit tests to a minimal 90%.

### PR DESCRIPTION
**Pull request by drrest**
## What
added: UnitTests for Source Okta - coverage 100%

## Recommended reading order
2. `y.python`

## 🚨 User Impact 🚨
No breaking changes

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>Updating a connector</strong></summary>

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-okta/unit_tests/test_source.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    GroupMembers,
+    GroupRoleAssignments,
+    Groups,
+    Logs,
+    OktaOauth2Authenticator,
+    SourceOkta,
+    UserRoleAssignments,
+    Users,
+)
+from unit_tests import (
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-okta/unit_tests/test_source.py:19]: Use `pytest.fixture` decorator to reuse some values.

File: airbyte-integrations/connectors/source-okta/unit_tests/test_source.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    def test_init_oauth2_authentication_wrong_oauth_config_bad_auth_type(self):
+        config = wrong_oauth_config_bad_auth_type()
+        source_okta = SourceOkta()
+        try:
+            source_okta.initialize_authenticator(config=config)
+        except Exception as e:
+            assert e.args[0] == "Config validation error. `auth_type` not specified."
+
+    def test_check_connection_ok(self, requests_mock):
+        config = oauth_config()
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-okta/unit_tests/test_source.py:62]: Try to use fixture to correspond DRY principle

**Review by lazebnyi — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-okta/unit_tests/__init__.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
+#
```
[the remark this answers is not in the document]
Comment (drrest) [reply airbyte-integrations/connectors/source-okta/unit_tests/__init__.py]: done

File: airbyte-integrations/connectors/source-okta/unit_tests/test_source.py (hunk 1 above)
Comment (drrest) [reply airbyte-integrations/connectors/source-okta/unit_tests/test_source.py]: done

File: airbyte-integrations/connectors/source-okta/unit_tests/test_source.py (hunk 2 above)
Comment (drrest) [reply airbyte-integrations/connectors/source-okta/unit_tests/test_source.py]: done

**Review by arsenlosenko — APPROVED** [review]

**Review by bazarnov — APPROVED** [review]

File: airbyte-integrations/connectors/source-okta/unit_tests/test_source.py [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+)
+
+
+class TestAuthentication:
+
+    def test_init_token_authentication_init(self, token_config, auth_token_config):
+        config = token_config
+        source_okta = SourceOkta()
+        token_authenticator_instance = source_okta.initialize_authenticator(config=config)
+        assert isinstance(token_authenticator_instance, TokenAuthenticator)
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-okta/unit_tests/test_source.py:27]: No need resign test config to config variable.

```suggestion
        source_okta = SourceOkta()
        token_authenticator_instance = source_okta.initialize_authenticator(config=token_config)
        assert isinstance(token_authenticator_instance, TokenAuthenticator)
```

The same for other tests.

**Review by lazebnyi — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-okta/unit_tests/test_source.py (hunk 4 above)
Comment (drrest) [reply airbyte-integrations/connectors/source-okta/unit_tests/test_source.py]: done

**Review by grubberr — APPROVED** [review]

File: airbyte-integrations/connectors/source-okta/unit_tests/test_streams.py [hunk 5]
```diff
@@ -0,0 +1,370 @@
...
+        expected_token = None
+        result = stream.next_page_token(**inputs)
+        assert result == expected_token
+
+
+class TestStreamUsers:
+
+    def test_stream_users(self, requests_mock, patch_base_class, users_instance, url_base, api_url):
+        stream = Users(url_base=url_base)
+        record = users_instance
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-okta/unit_tests/test_streams.py:175]: I'm not sure that we need resign here variable. 

Check please other tests.

**Review by lazebnyi — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-okta/unit_tests/test_streams.py (hunk 5 above)
Comment (drrest) [reply airbyte-integrations/connectors/source-okta/unit_tests/test_streams.py]: changed in others too
done

**Review by lazebnyi — APPROVED** [review]

**Comment by girarda** [comment]
This issue was closed because it was deprioritized. Feel free to reopen if needed.

**Pull request merged**